### DISCUSSION TITLE
What are module attributes, and how do they work when running in Deno?

### DISCUSSION THREAD

**Discussion by itsdouges** [denoland/deno#17138] [Q&A] [answered]
Heya! I've seen some modules on deno_x have some attributes such as "comes with Deno config". Does that mean it comes with the deno.json file? Can modules also ship their own import map? How does this affect runs from remote? Is there any documentation on it I can read?

**Comment by bartlomieju** [comment]
@crowlKats please take a look, I don't know how these attributes are generated.

> Can modules also ship their own import map? How does this affect runs from remote?

Import maps are meant to be used by applications, not libraries - they cannot be concatenated or "cascaded". If you want to ship and import map your users will have to manually update their own import map to include entries from your lib's import map.

**Comment by crowlKats** [comment, marked as answer]
These attributes are just a metric, not something functionally effective. there are currently 3 attributes:

- popularity metric: shows on a star-based rating the popularity of a module. 1 star means its in the top 10%, 2 stars in the top 5% and 3 stars in the top 1%.
- "By Deno Team": means the module is officially maintained by the deno team (meaning its under the denoland org)
- "comes with Deno config": means the module uses a deno config file. this is just a quality metric

**Reply by itsdouges** [reply]
Ah okay thanks mate, this is the answer. Would be good to expose some of this info in the UI for curious folk such as myself 😄, maybe as a tooltip or something.
